src/stat.h

```diff
@@ -483,10 +483,10 @@ template<class NUM> class MinValMax
   // happens when results are moved from a faster series to a slower one.
   MinValMax& operator+=(const MinValMax& rhs)
     {
-      //if (rhs.max > max) max=rhs.max;
-      //if (rhs.min < min) min=rhs.min;
-      max += rhs.max;
-      min += rhs.min;
+      if (rhs.max > max) max=rhs.max;
+      if (rhs.min < min) min=rhs.min;
+//      max += rhs.max;
+//      min += rhs.min;
       val += rhs.val;
```

Comment:
You are right that my proposed change doesn't make sense.  I am recording my thoughts on this here for later.

Each data sample covers a certain time period T and these time periods are joined to form more granular data of larger time periods.  So we have a set of time periods T0, T1, T2, ...

I think that the way the code is currently written, the min and max of a data point in Tn will be the min and max of the Tn-1 data points, recursively, so ultimately its the min/max of the T0 data points.

You are right that my proposed change is not useful, because if one of my "if" statements is hit, the "val" will have the same information as the min or max, resulting in loss of information.

But is there value in having it reflect just the Tn-1 intervals?  Depending on what is being accumulated the max/min value in a 10 second interval may be meaningless, but the average min/max over a longer time period may be.  In the case of the mempool, the statistic isn't added, its just sampled so the defined min/max makes a lot of sense.  But in the case of an accumulated statistic like "bytes sent", knowing the max over 10 seconds may be less valuable than knowing it per hour, when looking at the daily chart.  

This confusion is why I originally had the code written as you propose, but then changed it.  I am going to accept this change, but we may need to make changes to allow the user to specify the behavior of the min/max accumulation.